### DISCUSSION TITLE
Exclude directory while using --recursive option

### DISCUSSION THREAD

**Discussion by massimiliano96** [terraform-linters/tflint#2126] [Q&A]
Hi all!

I have the following repository structure for a Terraform module:

module                                                                                                                                                                       
└── examples
└── submodule1
└── submodule2

I would like to scan it using TFLint with the **recursive** option, in order to scan also the submodules.
How can I exclude the **examples/** folder?

**Comment by wata727** [comment]
Currently, it is not possible to exclude specific directories using the `--recursive` option. There are some workarounds:

- Implementing recursive directory walk yourself.
- Place `.tflint.hcl` with `disabled_by_default = true` in the `examples` directory.

This issue has been discussed before, but it remains an open problem with no optimal interface yet found.
Please take a look at the discussions and proposals below and if you have any feedback, please let us know.

- https://github.com/terraform-linters/tflint/discussions/1920
- https://github.com/terraform-linters/tflint/issues/1929
